### ISSUE TITLE
Improve error handling: Primary key longer than 512 bytes causes MDB_BAD_VALSIZE

### ISSUE THREAD

**Issue by queer** [meilisearch/meilisearch#4843]
**Describe the bug**
tldr: Primary key values longer than 512 bytes.

When my search index was being updated today, I got alerts that some Meilisearch tasks were failing. The following error came up repeatedly:

```
index_scheduler: Batch failed internal: MDB_BAD_VALSIZE: Unsupported size of key/DB name/data, or wrong DUPFIXED size.
```

I found that this was caused by primary keys being longer than 512 bytes. This is because some of my primary keys were autogenerated and I didn't know about this limitation.

Using `MEILI_EXPERIMENTAL_MAX_NUMBER_OF_BATCHED_TASKS` caused a significantly smaller part of the index to fail insertion, preserving much of the desired functionality while the issue was being fixed.

**To Reproduce**
Insert an object where the primary key is longer than 512 bytes. Large batch sizes seem to cause the entire batch to not write if this is hit.

**Expected behavior**
A more-useful warning when keys are too long.

**Meilisearch version:**
`1.9.0-1` via Ubuntu package. This also occurred on `1.8.3-1`. Running in Docker does not seem to matter.

**Comment by irevoire** [comment]
Hey, @queer, thanks for your reports. You're absolutely right.

Right now, I don't think we want to allow the usage of keys longer than 512 bytes, but we should definitely improve the error handling on this side and document it:
- [ ] Write about this limitation in the [documentation](https://www.meilisearch.com/docs/learn/advanced/known_limitations)
- [ ] Catch the error in the `enrich_documents` pipeline [here](https://github.com/meilisearch/meilisearch/blob/866922ecc3dfd1f40fe751561d756406f7ab719c/milli/src/update/index_documents/enrich.rs#L156) maybe to only fail the concerned task instead of the whole batch

**Comment by curquiza** [comment]
For people wanting to contribute: this issue is now about improving th error handling management of this error 👆

**Comment by flevi29** [comment]
I would like to contribute. So the idea here is to check early if the ID size is 512 bytes at most, return a specific error for it, and skip that one document instead of failing the whole batch? So far I have made it a general `InvalidDocumentId` like so:

```rs
fn validate_document_id(document_id: &str) -> Option<&str> {
    if !document_id.is_empty()
        && document_id.len() <= 512 // <-- this
        && document_id.chars().all(|c| matches!(c, 'a'..='z' | 'A'..='Z' | '0'..='9' | '-' | '_'))
    {
        Some(document_id)
    } else {
        None
    }
}

pub fn validate_document_id_value(document_id: Value) -> StdResult<String, UserError> {
    match document_id {
        Value::String(string) => match validate_document_id(&string) {
            // Why are we doing this? Isn't `s` the string slice contained in `string`, so basically one and the same?
            Some(s) if s.len() == string.len() => Ok(string),
            Some(s) => Ok(s.to_string()),
            None => Err(UserError::InvalidDocumentId { document_id: Value::String(string) }),
        },
        // a `u64` or `i64` cannot be more than 512 bytes once converted to a string
        Value::Number(number) if !number.is_f64() => Ok(number.to_string()),
        content => Err(UserError::InvalidDocumentId { document_id: content }),
    }
}
```

So I should rather return a specific error for that one case, and handle it here separately:
https://github.com/meilisearch/meilisearch/blob/866922ecc3dfd1f40fe751561d756406f7ab719c/milli/src/update/index_documents/enrich.rs#L147-L173

And then handle it here so that it doesn't fail the whole batch?
https://github.com/meilisearch/meilisearch/blob/866922ecc3dfd1f40fe751561d756406f7ab719c/milli/src/update/index_documents/enrich.rs#L107-L130

**Comment by flevi29** [comment]
> * [x]  Write about this limitation in the [documentation](https://www.meilisearch.com/docs/learn/advanced/known_limitations) - [Update known_limitations.mdx documentation#2925](https://github.com/meilisearch/documentation/pull/2925)
> 
> * [ ]  Catch the error in the `enrich_documents` pipeline [here](https://github.com/meilisearch/meilisearch/blob/866922ecc3dfd1f40fe751561d756406f7ab719c/milli/src/update/index_documents/enrich.rs#L156) maybe to only fail the concerned task instead of the whole batch

@irevoire For any invalid id the whole batch was aborted, if I understand correctly. Same for invalid geo and maybe something else. It feels to me like it should be a separate issue to partially fail a batch operation.

There's also this fixme, what is the intention here more specifically? I could do it.
https://github.com/meilisearch/meilisearch/blob/866922ecc3dfd1f40fe751561d756406f7ab719c/milli/src/documents/primary_key.rs#L150

So is it enough to adjust the error message and return an error as described above?
